src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java

```diff
@@ -37,61 +48,23 @@ public List<OrderDetails> getAllOrders() {
 
         List<OrderDetails> list = new ArrayList<>(orders.size());
         for (Order order : orders) {
-            OrderDetails orderDetails = collectOrderDetails(order);
+            OrderDetails orderDetails = orderMapper.toOrderDetails(order);
+            orderDetails.setTotalPrice(getTotalPriceForOrder(orderDetails.getOrderItems()));
+            orderDetails.setOrderItems(collectOrderItemDetails(order.getId()));
             list.add(orderDetails);
         }
         return list;
     }
 
-
-    private OrderDetails collectOrderDetails(Order order) {
-        OrderDetails orderDetails = new OrderDetails();
-        orderDetails.setId(order.getId());
-        orderDetails.setOrderItems(collectOrderItemDetails(order.getId()));
-        orderDetails.setCustomerId(order.getUserId());
-        orderDetails.setUpdatedAt(order.getUpdatedAt());
-        orderDetails.setCreatedAt(order.getCreatedAt());
-        orderDetails.setTotalPrice(getTotalPriceForOrder(orderDetails.getOrderItems()));
-        return orderDetails;
-    }
-
-    private BigDecimal getTotalPriceForOrder(List<OrderItemDetails> orderItems) {
-        BigDecimal result = BigDecimal.ZERO;
-        for (OrderItemDetails orderItem : orderItems) {
-            BigDecimal totalPrice = orderItem.getProduct().getPrice()
-                    .multiply(BigDecimal.valueOf(orderItem.getQuantity()));
-            result = result.add(totalPrice);
-        }
-        return result;
-    }
-
     private List<OrderItemDetails> collectOrderItemDetails(Integer orderId) {
         return orderItemRepository.getOrderItemsByOrderId(orderId).stream()
-                .map(orderItem -> convertOrderItemToOrderItemDetails(orderItem))
+                .map(orderItem -> orderItemMapper.toOrderItemDetails(orderItem,
+                        getProductById(orderItem.getProductId())))
                 .toList();
```

Comment:
I recommend doing it in a slightly different way: you can fetch all OrderItems by Order, then iterate the collection with all OrderItems and collect all product IDs, then pass these product IDs to ProductRepository to fetch all Products by IDs **in one request**.

This will allow you to decrease the load on ProductRepository.

Also, please think what data structures you can use in this new approach.

Reply:
refactored to load from product repository Map, where key is productId and value is product object